lab/tree_builder.py

```diff
@@ -13,21 +14,35 @@ def __init__(self, name, val):
 class TreeBuilder(object):
 
     def __init__(self, root):
-        # root will be type TreeNode
         self.root = root
         self.size = 0
 
     # Function to push to tree 
     def push(self, node):
-        # TODO
-        pass
+        currentNode = self.root
+        while True:
```

Comment:
This will incorrectly add an extra node if the player node already exists in the tree. Let's handle that edge case by doing a check at the beginning to see if the node exists in the tree and return a boolean to signify if the node was successfully added or not.